package.json

```diff
@@ -615,6 +615,7 @@
 	"scripts": {
 		"compile": "tsc -p ./",
 		"watch": "tsc -p -watch ./",
+		"pretest": "winCompile.bat",
```

Comment:
this is windows only; for mac it's ./macCompile.sh
I don't know what's the best way to make it cross platform, but I found the same question here:
https://stackoverflow.com/questions/45082648/npm-package-json-os-specific-script

Reply:
Using python or JS to do this job.

Reply:
just corious, winCompile.bat looks not used anywhere. what is this file used for?

Reply:
after reading the code briefly, I don't see any need for the unit test to call this command. Anything I was missing?

Reply:
If you want to run unit tests on any of the Help related functions, you'll need to make sure the `webHelpAbstraction.json` file gets copied where it belongs somehow, but other than that the `acadProcessFinder.exe` copy can't be of any use in CI.

I had suggested we set these copy operations up using a python script hooked into the tasks.json a while back, but I am not sure if that will play well with your plans. I just know its worked pretty well for me out of CI.
https://github.com/JD-Howard/FabCOD/blob/main/.vscode/tasks.json

Reply:
@LanDong-adsk winCompile.bat (and ./macCompile.sh) are used for developers to do a clean build, i.e. to clean up the out folder first, and then npm run compile.
Long ago we got some weird problem on dev machine, which was gone after locally doing a clean build. Then we added these two scripts.
I once wanted to explain the use of these scripts in DevReadme, but Hualin stopped me because he thought that the new comers could be confused by too much information.

Reply:
winCompile.bat is necessary to run test especially during the test cases writing since the changed and compiled script will impact the result and I have seen some problems related to this recently.
But for Github CI enviornment since the env is clean it is unnecessary to run it.